pkg/kubelet/apis/cri/runtime/v1alpha2/api.proto

```diff
@@ -997,6 +997,13 @@ message Image {
     // User name that will run the command(s). This is used if UID is not set
     // and no user is specified when creating container.
     string username = 6;
+    // GID that will run the command(s). This is used as a default if no group is
+    // specified when creating the container. GID and the following group name
+    // are mutually exclusive.
+    Int64Value gid = 7;
+    // Group name that will run the command(s). This is used if GID is not set
+    // and no group is specified when creating container.
+    string groupname = 8;
```

Comment:
Is groupname actually needed? Can't you just ensure that the gid is not 0? Or is it possible for gid to return as non-zero, but still have a groupname set?

Reply:
When Dockerfile has USER specified as non numeric then groupname will contain that non numeric group name. This is mainly being used to throw an error message saying, we cannot determine if the groupname is zero or not since its non numeric . See similar error being thrown here https://github.com/kubernetes/kubernetes/blob/f634f7dae4a49bb513ae8f5c53e5d115308231d3/pkg/kubelet/kuberuntime/security_context.go#L92